Realm/Realm.Weaver/Analytics/Analytics.cs

```diff
@@ -0,0 +1,846 @@
+                {
+                    if (!(member is PropertyDefinition property))
+                    {
+                        return false;
+                    }
+
+                    if (((GenericInstanceType)property.PropertyType).GenericArguments[0].IsPrimitive)
+                    {
+                        featureDict[PrimitiveSet] = 1;
+                    }
+                    else
+                    {
+                        featureDict[ReferenceSet] = 1;
+                    }
+
+                    return featureDict[PrimitiveSet] == 1 &&
+                        featureDict[ReferenceSet] == 1;
+                }
+            },
+            {
+                "RealmInteger`1", (member, featureDict, references) =>
+                {
+                    featureDict[RealmInteger] = 1;
+                    return true;
+                }
+            },
+        };
+
+        #endregion
+
+        private readonly Config _config;
+
+        internal Analytics(Config config, ImportedReferences references, ILogger logger)
+        {
+            _references = references;
+            _config = config;
+            _logger = logger;
+        }
+
+        private void AnalyzeRealmClass(TypeDefinition type)
+        {
+            Func<IMemberDefinition, Dictionary<string, byte>, ImportedReferences, bool> featureFunc = null;
+
+            if (_classAnalysisSetters.TryGetValue(nameof(IAsymmetricObject), out featureFunc) &&
+                (type.IsIAsymmetricObjectImplementor(_references) || type.IsAsymmetricObjectDescendant(_references)) &&
+                featureFunc.Invoke(type, _realmFeaturesToAnalyse, _references))
+            {
+                _classAnalysisSetters.Remove(nameof(IAsymmetricObject));
+            }
+            else if (_classAnalysisSetters.TryGetValue(nameof(IEmbeddedObject), out featureFunc) &&
+                (type.IsIEmbeddedObjectImplementor(_references) || type.IsEmbeddedObjectDescendant(_references)) &&
+                featureFunc.Invoke(type, _realmFeaturesToAnalyse, _references))
+            {
+                _classAnalysisSetters.Remove(nameof(IEmbeddedObject));
+            }
+
+            foreach (var property in type.Properties)
+            {
+                var key = property.PropertyType.Name;
+                if (!_classAnalysisSetters.ContainsKey(key))
+                {
+                    // when looking for "Class" type
+                    key = property.PropertyType.MetadataType.ToString();
+                }
+
+                if (_classAnalysisSetters.TryGetValue(key, out featureFunc) &&
+                    featureFunc.Invoke(property, _realmFeaturesToAnalyse, _references))
+                {
+                    // if the byte is set, remove the entry from the dict to avoid future unnecessary work
+                    _classAnalysisSetters.Remove(key);
+                }
+            }
+        }
+
+        internal void AnalyzeUserAssembly(ModuleDefinition module)
```

Comment:
Small thing, but sometimes you use analyse, some other analyze. I noticed because I couldn't find some methods with search 😁